overlord/auth/auth.go

```diff
@@ -117,6 +131,88 @@ func User(st *state.State, id int) (*UserState, error) {
 	return nil, fmt.Errorf("invalid user")
 }
 
+// SetDeviceIdentity sets the device's identity as used for communication with the store.
+// XXX: Should eventually require the serial assertion to be the
+// assertions DB rather than taking it directly, but that requires
+// significant work.
+// XXX: Doesn't check that the serial assertion is trustworthy, just
+// that it matches the purported identity. But the assertion is only
+// used to establish a store session, and the store performs the
+// requisite checks.
+func SetDeviceIdentity(st *state.State, brand string, model string, serial string, serialAssertionEncoded []byte) error {
+	var authStateData AuthState
+
+	// Verify that the serial assertion matches the brand, model and
+	// serial.
+	assert, err := asserts.Decode(serialAssertionEncoded)
+	if err != nil {
+		return err
+	}
+	if assert.Type() != asserts.SerialType {
+		return fmt.Errorf("serial assertion is actually %s", assert.Type())
+	}
+	serialAssertion := assert.(*asserts.Serial)
+	if brand != serialAssertion.BrandID() || model != serialAssertion.Model() || serial != serialAssertion.Serial() {
```

Comment:
We need to define when and how we want to validate that info. Seems okay to not do that here to begin with. We'll definitely want to validate it over the run time of the system rather than just when setting as well.